grudge/trace_pair.py

```diff
@@ -318,8 +319,20 @@ def interior_trace_pair(dcoll: DiscretizationCollection, vec) -> TracePair:
     return local_interior_trace_pair(dcoll, vec)
 
 
+@dataclass(frozen=True)  # for KeyBuilder support
+class CommTag:
+    """A communication tag with a hash value that is stable across
+    runs, even without setting ``PYTHONHASHSEED``."""
+    @memoize_method
+    def __hash__(self):
+        return hash(tuple(str(type(self)).encode("ascii")))
+
+    def __eq__(self, other):
+        return isinstance(other, type(self))
```

Comment:
This is incorrect: It accepts arbitrary subclasses for `other`, and violates the property of "`__eq__` implies hash equality."

Reply:
This is (hopefully) fixed in [6918c63](https://github.com/inducer/grudge/pull/319/commits/6918c63ca6a69829ed9bfb805045657020a2187e)